src/main/java/com/spotify/avro/mojo/AvscFilesCompiler.java

```diff
@@ -95,6 +96,9 @@ private boolean tryCompile(AvroFileRef src, File outputDirectory) {
     compiler.setFieldVisibility(fieldVisibility);
     compiler.setEnableDecimalLogicalType(enableDecimalLogicalType);
     compiler.setCreateSetters(createSetters);
+    compiler.setGettersReturnOptional(optionalGetters);
+    compiler.setOptionalGettersForNullableFieldsOnly(optionalGetters);
```

Comment:
Not sure if we should use the same flag for this. We might want `Optional<T>` for "nullable" fields only and `T` for "required" fields?

Reply:
Actually I did exactly that in this PR: https://github.com/apache/avro/pull/965 , but I changed my mind in this PR since I don't think it's useful to have Optional<T> for non nullable fields, or am I'm missing something here?

Reply:
Ah sorry i misunderstood the flag. It does the opposite of what I thought. We can keep this as it.